### PR TITLE
[systemtest][azure] Upgrade pipelines

### PR DESCRIPTION
**Pull request by im-konge**
Signed-off-by: Lukas Kral <lukywill16@gmail.com>

### Type of change

- Bugfix

### Description

This PR fixes the upgrade tests for Azure pipelines.

1) First problem was that TO and UO were overrided by different images from the CO
2) Second problem was that we firstly changed the `DOCKER_ORG`, `DOCKER_TAG` and `DOCKER_IREGISTRY` in `StrimziUpgradeST.json` and `060..` and then we builded the images -> this two files were (again) overriden by the `mvn clean install` build
3) The docker tag, org and registry wasn't correctly setup
4) `STRIMZI_IMAGE_PULL_POLICY` was unfortunately set by default to `Always`, so all the tests where the `:latest` image was used tried to pull the images from docker.hub, but it failed

### Checklist

- [x] Make sure all tests pass

### CODE REVIEW

**Review by scholzj — APPROVED** [review]

File: .azure/templates/system_test_general.yaml [hunk 1]
```diff
@@ -30,20 +30,21 @@ jobs:
...
-    - bash: ".azure/scripts/setup_upgrade.sh"
-      displayName: "Setup environment for upgrade"
-      condition: eq('${{ parameters.groups }}', 'upgrade')
-
     - template: "general_steps.yaml"
 
     - bash: |
         make docker_build
         make docker_tag
+        docker images
```
Comment (Frawless) [inline .azure/templates/system_test_general.yaml:38]: leftover?

**Review by Frawless — APPROVED** [review]

File: .azure/templates/system_test_general.yaml (hunk 1 above)
Comment (im-konge) [reply .azure/templates/system_test_general.yaml:38]: Yes, but I was thinking that this is "good to have" there. Gonna remove it, thanks.

**Pull request merged**